src/main/java/com/mongodb/spark/sql/connector/MongoCatalog.java

```diff
@@ -178,12 +178,11 @@ public void alterNamespace(final String[] namespace, final NamespaceChange... ch
     throw new UnsupportedOperationException("Altering databases is currently not supported");
   }
 
-  /**
-   * Drop a database.
-   *
-   * @param namespace (database) a multi-part namespace
-   * @return true if the namespace (database) was dropped
-   */
+  @Override
+  public boolean dropNamespace(final String[] namespace, final boolean cascade) {
```

Comment:
`dropNamespace(namespace)` was removed in 3.4
`dropNamespace(namespace, cascade)` was added in 3.4 - probably with a default value for cascade.

Reply:
Scala allows default values for parameters - so they may have just done that.